src/test/java/net/sf/jsqlparser/statement/upsert/UpsertTest.java

```diff
@@ -68,7 +68,21 @@ public void testUpsertDuplicate() throws JSQLParserException {
         assertTrue(upsert.isUseDuplicate());
         assertEquals(statement, "" + upsert);
     }
-
+    @Test
+    public void testUpsertIgnore() throws JSQLParserException {
+        String statement = "UPSERT INTO TEST (ID, COUNTER) VALUES (123, 0) ON DUPLICATE KEY IGNORE";
+        Upsert upsert = (Upsert) parserManager.parse(new StringReader(statement));
```

Comment:
Introduce a deparser test as well using those assert that a statement can be parsed and deparsed methods from TestUtils.